imgui.lua

```diff
@@ -46,11 +47,23 @@ imgui.Hook("PreRender", "Input", function()
 	-- calculate mouse state
 	if shouldAcceptInput() then
 		local wasPressing = gState.pressing
-		gState.pressing = input.IsMouseDown(MOUSE_LEFT) or input.IsKeyDown(KEY_E)
+		gState.pressing = pressing[IN_USE] or pressing[IN_ATTACK]
```

Comment:
Instead of using a table and indexing it twice every frame you could use an "input counter" variable which counts the amount of acceptable inputs being held. This would a more efficient approach.

Reply:
In this case it's already more efficient than the original implementation, I'm reducing two calls and two indexes per frame is negligible.
I can add your micro-optimization if you'd like however.

Reply:
Turns out an input counter wouldn't work in multiplayer anyways. Seems to give some unexpected [results](https://i.imgur.com/Yw8o9qQ.png) when simply incrementing/decrementing per key press/release.

Reply:
Turns out KeyPress and KeyReleased are both predicted hooks, that's why you're receiving duplicate events.

Reply:
That explains the unintended behavior, thanks for the information. I'll be sure to keep that in mind for the future.